### PR TITLE
:sparkles: Consider the document embedded if isCctEmbedded is true.

### PR DESCRIPTION
**Pull request by ryjust**
This change allows CCT embedded documents to perform CSI logging and update the referrer when appropriate. This change also makes the condition for being considered CCT embedded slightly more strict.

### CODE REVIEW

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -245,14 +245,21 @@ export class Viewer {
         this.prerenderSize_;
     dev().fine(TAG_, '- prerenderSize:', this.prerenderSize_);
 
+    const queryParams = parseQueryString(this.win.location.search);
+    /**
+     * Whether the AMP document is embedded in a Chrome Custom Tab.
+     * @private @const {boolean}
+     */
+    this.isCctEmbedded_ = !this.isIframed_ && queryParams['amp_gsa'] === '1' &&
+        (queryParams['amp_js_v'] || '').startsWith('a');
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: You'll have to import the `startsWith` helper from [src/string.js](https://github.com/ryjust/amphtml/blob/428fc6e8730addd928342ccaa6e462a7926897e9/src/string.js#L72-L83)

File: src/service/viewer-impl.js [hunk 2]
```diff
@@ -273,15 +280,9 @@ export class Viewer {
             // Parent asked for viewer JS. We must be embedded.
             || (this.win.location.search.indexOf('amp_js_v') != -1))
         || this.isWebviewEmbedded_
+        || this.isCctEmbedded_
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: This also affects viewer messaging. Need to confirm that you setup the viewer origin, referral url, viewer url and that you actually setup messaging.

Comment (ryjust) [reply src/service/viewer-impl.js]: Yes, I updated the AGSA integration script already and will be rolling it out over the next week. It should be at 100% on Wednesday.

File: src/service/viewer-impl.js (hunk 1 above)
Comment (ryjust) [reply src/service/viewer-impl.js]: Imported!

File: src/service/viewer-impl.js [hunk 3]
```diff
@@ -245,14 +246,21 @@ export class Viewer {
         this.prerenderSize_;
     dev().fine(TAG_, '- prerenderSize:', this.prerenderSize_);
 
+    const queryParams = parseQueryString(this.win.location.search);
```
Comment (choumx) [inline src/service/viewer-impl.js]: We previously had this inlined to avoid unnecessary work when `this.isIframed_ == true`.

Comment (ryjust) [reply src/service/viewer-impl.js]: Updated so that it doesn't do the work when this.isIframed == true

File: src/service/viewer-impl.js [hunk 4]
```diff
@@ -253,14 +254,25 @@ export class Viewer {
         this.prerenderSize_;
     dev().fine(TAG_, '- prerenderSize:', this.prerenderSize_);
 
+    let isCctEmbedded = false;
+    if (!this.isIframed_) {
+      const queryParams = parseQueryString(this.win.location.search);
```
Comment (choumx) [inline src/service/viewer-impl.js]: Nit: `searchParams` would be more accurate, though we can fix this later.

**Pull request merged**